core/src/main/java/de/jplag/strategy/ParallelComparisonStrategy.java

```diff
@@ -28,8 +31,13 @@ public JPlagResult compareSubmissions(SubmissionSet submissionSet) {
         }
 
         List<SubmissionTuple> tuples = buildComparisonTuples(submissionSet.getSubmissions());
-        List<JPlagComparison> comparisons = tuples.stream().parallel().map(tuple -> compareSubmissions(tuple.left(), tuple.right()))
-                .flatMap(Optional::stream).toList();
+        ProgressBar progressBar = ProgressBarLogger.createProgressBar(ProgressBarType.COMPARING, tuples.size());
+        List<JPlagComparison> comparisons = tuples.stream().parallel().map(tuple -> {
+            Optional<JPlagComparison> result = compareSubmissions(tuple.left(), tuple.right());
+            progressBar.step();
```

Comment:
Is there a reason this is done here and not in the compareSubmissions method? This ties the logging to the parallel strategy and makes the code a bit more cluttered.

Reply:
As the progress bar behaves like a logger, I would create it in `AbstractComparisonStrategy` and store it as a (private) field. I would then call `progressBar.step()`  after the GST in  `AbstractComparisonStrategy.compareSubmissions()`. The advantage would be, that the concrete strategy never needs to bother with the progress bar. You can even get rid of the the calculation if you create the progress bar in the `buildComparisonTuples()` method.